src/app/shell/personal-cabinet/shared-cabinet/user-config/user-config-edit/user-config-edit.component.ts

```diff
@@ -124,6 +123,11 @@ export class UserConfigEditComponent extends CreateFormComponent implements OnIn
   }
 
   public onCancel(): void {
-    this.location.back();
+    this.navigationPaths$
+      .subscribe((value) => {
+        const path = value[value.length - 2].path ?? '/personal-cabinet/config';
+        this.router.navigate([path]);
+      })
+      .unsubscribe();
```

Comment:
Other solution is using hard navigation to parent page like it is in CreateProviderComponent:
`this.router.navigate(['/personal-cabinet/provider/info']);`
In our case it will look like:
`this.router.navigate(['/personal-cabinet/config]);`

Reply:
maybe we can play with this or the old school location back in [ConfirmationModalWindowComponent](https://github.com/ita-social-projects/OoS-Frontend/blob/develop/src/app/shared/components/confirmation-modal-window/confirmation-modal-window.component.ts).
Something like if type of modal is `ModalConfirmationType.leavePage` then the confirm button on the modal will navigate back. That way we probably can modify only the modal code so it will work in the entire project and think about some other option to trigger the candeactivate guard in `onBack()` instead of `location.back()`

Reply:
Do we actually need to use `location.back()`? Maybe we should make navigation like in [CreateProviderComponent](src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts) - just to parent route?
I think no one is going to directly navigate to form create/edit URL for example from other website, edit info and then return to that website instead of seeing results of actions.
Anyway, I haven't found approaches to manually trigger `CreateGuard.canDeactivate()` without routing, also there is option for injecting `CreateGuard`, but it's many code for each component with form. I'm suggesting use `this.router.navigate[]` for parent routes

Reply:
As for me, I see 2 possible solutions:
1) use a direct route like `this.router.navigate(['/personal-cabinet/config]);`
2) since `onCancel` or `onBack` are used in many forms, you can try to figure out a more general solution, it could be something like this `this.router.navigate(['../'], { relativeTo: this.route });`, but it requires more local testing to make sure everything works as expected. 

So if you have time, you can test the second option for other cases, or we can replace `this.location.back()` with `this.router.navigate(['/personal-cabinet/config]);`.

Reply:
> As for me, I see 2 possible solutions:
> 
> 1. use a direct route like `this.router.navigate(['/personal-cabinet/config]);`
> 2. since `onCancel` or `onBack` are used in many forms, you can try to figure out a more general solution, it could be something like this `this.router.navigate(['../'], { relativeTo: this.route });`, but it requires more local testing to make sure everything works as expected.
> 
> So if you have time, you can test the second option for other cases, or we can replace `this.location.back()` with `this.router.navigate(['/personal-cabinet/config]);`.

What about using `routerLink` attribute in HTML template for components? I have tested it locally - it doesn't break navigation history and calling `CreateGuard` well. Since `onCancel` or `onBack` methods don't have any additional logics (in cases where they are) I suggest using `routerLink`, what do you think?

Reply:
The `routerLink` does the same thing as `this.router.navigate`, so both solutions are good. 
I prefer the solution of keeping the `onCancel` method, and we can add it as an abstract to `CreateFormComponent` (and `onSubmit` too) because in every form that implements this interface we have this logic.